src/periphery/MgvReader.sol

```diff
@@ -316,4 +322,74 @@ contract MgvReader {
       /* end if success branch **/
     }
   }
+
+  /* Open markets tracking */
+
+  // MgvMarketTracker lets anyone query for a market state (open or not), and
+  // gets the list of open markets.
+
+  // Anyone can permisionlessly refresh the state of a market with updateMarket(tkn0,tkn1)
+
+  // Note that **no check on the actual Mangrove market state is performed at read time**. It
+  // is the DAO's responsibility to accurately reflect the state of Mangrove
+  // markets.
+
+  // array of all open markets
+  function openMarkets() external view returns (address[2][] memory) {
+    return _openMarkets;
+  }
+
+  function numOpenMarkets() external view returns (uint) {
+    return _openMarkets.length;
+  }
+
+  // slice of open markets, starting at `from`, of length `maxLen` or shorter if overflow
+  // revert if `from > numMarkets`
+  function openMarkets(uint from, uint maxLen) external view returns (address[2][] memory) {
+    uint numMarkets = _openMarkets.length;
+    if (from + maxLen > numMarkets) {
+      maxLen = numMarkets - from;
+    }
+    address[2][] memory openMarketsSlice = new address[2][](maxLen);
+    unchecked {
+      for (uint i = 0; i < maxLen; i++) {
+        openMarketsSlice[i] = _openMarkets[from + i];
+      }
+    }
+    return openMarketsSlice;
+  }
+
+  // canonicalize ordering
+  function order(address tkn0, address tkn1) public pure returns (address, address) {
+    return uint160(tkn0) < uint160(tkn1) ? (tkn0, tkn1) : (tkn1, tkn0);
+  }
+
+  function isMarketOpen(address tkn0, address tkn1) external view returns (bool) {
```

Comment:
Consider using natspec so that we can easily show it in our docs.mangrove.exchange.